src/ssl/gadgets.cc

```diff
@@ -421,7 +414,11 @@ mimicExtensions(Security::CertPointer & cert, Security::CertPointer const &mimic
     // XXX: Add PublicKeyPointer. In OpenSSL, public and private keys are
     // internally represented by EVP_PKEY pair, but GnuTLS uses distinct types.
     const Security::PrivateKeyPointer certKey(X509_get_pubkey(mimicCert.get()));
-    const auto rsaPkey = EVP_PKEY_get0_RSA(certKey.get()) != nullptr;
+#if OPENSSL_VERSION_MAJOR < 3
+    const auto rsaPkey = bool(EVP_PKEY_get0_RSA(certKey.get()));
```

Comment:
I think the original code relays the intent better. There is no need to change it here AFAICT:

```suggestion
    const auto rsaPkey = EVP_PKEY_get0_RSA(certKey.get()) != nullptr;
```